lib/std/Build.zig

```diff
@@ -409,45 +409,48 @@ fn createChildOnly(
 }
 
 fn userInputOptionsFromArgs(allocator: Allocator, args: anytype) UserInputOptionsMap {
-    var user_input_options = UserInputOptionsMap.init(allocator);
+    var user_input_options: UserInputOptionsMap = .init(allocator);
     inline for (@typeInfo(@TypeOf(args)).@"struct".fields) |field| {
-        const v = @field(args, field.name);
-        const T = @TypeOf(v);
-        switch (T) {
+        const maybe_value = @field(args, field.name);
+        const given, const value = switch (@typeInfo(@TypeOf(maybe_value))) {
+            .optional => if (maybe_value) |v| .{ true, v } else .{ false, undefined },
+            else => .{ true, maybe_value },
+        };
+        if (given) switch (@TypeOf(value)) {
```

Comment:
Also, what should the behavior be when just specifying `null` instead of something like `@as(?std.builtin.OptimizeMode, null)` as in your example? With both my suggested change and your current code, I believe this would just be a compile error (null has it's own type; it's not covered by the `.optional` case). I think it may be better to just treat plain null as not specified to be consistent with optionals.

Example fix:

```zig
inline for (@typeInfo(@TypeOf(args)).@"struct".fields) |field| blk: {
    const maybe_value = @field(args, field.name);
    const value = switch (@typeInfo(@TypeOf(maybe_value))) {
        .optional => maybe_value orelse break :blk,
        .null => break :blk,    // <-- Note this added case
        else => maybe_value,
    };
    switch (@TypeOf(value)) {
        ...
```